pkg/api/auth_middleware_test.go

```diff
@@ -12,8 +12,16 @@ import (
 	"github.com/treeverse/lakefs/pkg/auth/model"
 )
 
-func TestAuthMiddleware(t *testing.T) {
```

Comment:
How was the `AuthMiddleware` impacted by this PR? 
The downside for testing everything twice is the runtime of the tests. It's not a big deal though..

Reply:
Most of the tests in the api package are not impacted by the changes. I introduced the kv flavor to tests using the hookHandler to make sure nothing got accidentally broken.